source/uwp/Renderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -533,12 +489,162 @@ namespace AdaptiveCards::Rendering::Uwp
+                imageContainerRef = winrt::make_weak(imageContainer), isVisible, streamRef = winrt::make_weak(stream),
+                imageSourceRef = winrt::make_weak(imageSource.as<winrt::SvgImageSource>()), renderContext](
+                    auto const& result, auto status) -> void
+                {
+                    if (auto strongThis = weakThis.get())
+                    {
+                        if (status == winrt::AsyncStatus::Completed)
+                        {
+                            auto bytes = result.GetResults();
+                            auto svgText = streamDataReader.ReadString(bytes);
+
+                            auto svgDocument = winrt::XmlDocument();
+                            svgDocument.LoadXml(svgText);
+
+                            strongThis->ParseXmlForHeightAndWidth(renderContext, svgDocument, imageSourceRef);
+                        }
+
+                        // Now that we've parsed the height, we can set the image source
+                        strongThis->SetSvgImageSourceAsync(
+                            renderContext, imageSourceRef, streamRef, uiElementRef, isAutoSize, parentElementRef, imageContainerRef, isVisible);
+                    }
+                });
+        }
+        else
+        {
+            auto setSourceAction = imageSource.as<winrt::BitmapImage>().SetSourceAsync(stream);
+
+            setSourceAction.Completed([weakThis = this->get_weak(), uiElement, isAutoSize, parentElement, imageContainer, isVisible](
+                auto const& /*operation*/, auto status)
+                {
+                    if (status == winrt::AsyncStatus::Completed && isAutoSize)
+                    {
+                        if (auto strongThis = weakThis.get())
+                        {
+                            strongThis->SetAutoSize(uiElement, parentElement, imageContainer, isVisible, false /* imageFiresOpenEvent */);
+                        }
+                    }
+                });
+        }
+    }
+
+    winrt::IAsyncAction render_xaml::XamlBuilder::SetSvgUriSource(winrt::AdaptiveRenderContext const& renderContext,
+                                                                  winrt::weak_ref<winrt::SvgImageSource> const& imageSourceRef,
+                                                                  winrt::weak_ref<winrt::Uri> const& uriRef)
+    {
+        try
+        {
+            auto imageSource = imageSourceRef.get();
+            auto uri = uriRef.get();
+
+            if (imageSource && uri)
+            {
+                co_await winrt::resume_foreground(imageSource.Dispatcher());
+                imageSource.UriSource(uri);
+            }
+        }
+        catch (winrt::hresult_wrong_thread ex)
+        {
+            renderContext.AddWarning(winrt::WarningStatusCode::AssetLoadFailed,
```

Comment:
I don't think this is kind of an error that we need to log in `renderContext` :)
executing code on the wrong thread is **our** fault in the code, not that the user or system has failed:)
I don't think we really need to try/catch exception here :) 
I don't see it failing if it's done correctly:)

Reply:
Okay, I'll remove the try/catch block